src/services/workflow.service.js

```diff
@@ -106,108 +245,131 @@ class WorkflowService extends GQuery {
     }
-    if (this.debug) {
-      // eslint-disable-next-line no-console
-      console.debug('graphql request:', this.query)
-    }
-    const vm = this
-    if (this.observable !== null) {
-      this.observable.unsubscribe()
-      this.observable = null
+
+    // We will use the first subscriber to compare its variables, and also will
+    // merge other queries into its base query, in-place.
+    const baseSubscriber = subscribers[0]
+
+    // Reset.
+    const initialQuery = print(baseSubscriber.query.query)
+    subscription.query.query = baseSubscriber.query.query
+    subscription.actionNames = baseSubscriber.query.actionNames
+    subscription.tearDownActionNames = baseSubscriber.query.tearDownActionNames
+
+    subscribers.slice(1)
+      .forEach(subscriber => {
+        if (!isEqual(subscriber.query.variables, baseSubscriber.query.variables)) {
+          throw new Error('Error recomputing subscription: Query variables do not match.')
+        }
+        this.mergeQueries(baseSubscriber.query.query, subscriber.query.query)
+        union(subscription.actionNames, subscriber.query.actionNames)
+        union(subscription.tearDownActionNames, subscriber.query.tearDownActionNames)
+      })
+    const finalQuery = print(baseSubscriber.query.query)
+    // TODO: consider using a better approach than print(a) === print(b)
+    // If we changed the query due to query-merging, then we know we must reload its
+    // GraphQL subscription (i.e. stop subscription, start a new one with the server).
+    if (initialQuery !== finalQuery) {
+      subscription.reload = true
     }
-    this.observable = this.apolloClient.subscribe({
-      query: this.query,
-      fetchPolicy: 'no-cache'
-    }).subscribe({
-      next (response) {
-        // commit results
-        store.dispatch(
-          'workflows/set',
-          response.data.workflows
-        )
-        // set all subscriptions to active
-        vm.subscriptions
-          .filter(s => s.active === false)
-          .forEach(s => {
-            s.active = true
-          })
-        // run callback functions on the views
-        vm.callbackActive()
-      },
-      error (err) {
-        store.dispatch(
-          'setAlert',
-          new Alert(err.message, null, 'error')
-        )
-      },
-      complete () {
-      }
-    })
+    // And here we set the new merged-query. Voila!
+    subscription.query.query = baseSubscriber.query.query
   }
 
-  /** Call a mutation on the given identifier.
-   *
-   * Will only work if no additional information (other than the identifier
-   * itself) is required.
-   *
-   * Requires an absolute identifier (i.e. must include the workflow name).
-   */
-  mutate (mutationName, id) {
-    const mutation = this.getMutation(mutationName)
-    return mutate(
-      mutation,
-      getMutationArgsFromTokens(
-        mutation,
-        tokenise(id)
-      ),
-      this.apolloClient
-    )
+  gClone (query) {
```

Comment:
This function, and the next functions down are all from the old `gquery.js`. I just moved them here. We are not changing how we merge queries, I've reverted that change, but will create disabled unit tests for cases where the current approach could result in bugs, or where it's missing some feature :+1: For now it should work fine for what we need.